src/sgame/sg_cmds.cpp

```diff
@@ -228,10 +227,10 @@ int G_ClientNumberFromString( const char *s, char *err, int len )
 
 		if ( strstr( n2, s2 ) )
 		{
-			if ( p )
+			if ( err )
 			{
-				l = Q_snprintf( p, l2, "%-2d — %s^*", i, cl->pers.netname );
-				p += l;
+				l = Q_snprintf( err, l2, "%-2d — %s^*", i, cl->pers.netname );
+				err += l;
 				l2 -= l;
 			}
 
```

Comment:
Original value of `err` is used at the end of the function

Reply:
Exact. I did not understood that, in fact, here an error messaged is prepared in case more than one approximative matches are found, which would then be overwritten if no match at all can be.